### PR TITLE
add the What's new tool

### PR DESCRIPTION
**Pull request by BillWagner**
Add the what's new tool and dependencies to the public repo.

Contributes to #99

### CODE REVIEW

File: WhatsNew.Cli/WhatsNew.Cli.csproj [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <Nullable>enable</Nullable>
+	  <ImplicitUsings>enable</ImplicitUsings>
```
Comment (Youssef1313) [inline WhatsNew.Cli/WhatsNew.Cli.csproj:7]: nit: Indentation wasn't correct

```suggestion
    <ImplicitUsings>enable</ImplicitUsings>
```

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+Build automation is handled by Azure Pipelines via the [dotnet-docs-tools build pipeline](https://dev.azure.com/mseng/TechnicalContent/_build?definitionId=10534). A YAML representation of this build pipeline is stored in the *azure-pipelines.yml* file.
+
+The build pipeline contains a `trigger` section which determines the branch names/patterns and paths that trigger a build. For example:
+
+```yml
+trigger:
+  branches:
+   include:
+     - master
```
Comment (IEvangelist) [inline WhatsNew.Cli/CONTRIBUTING.md:75]: Think this would be `main` now?
```suggestion
     - main
```

**Review by IEvangelist — APPROVED** [review]
Looks good to me, let's :shipit:.

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+﻿## Prerequisites
+
+The following components are required to run the tool locally:
+
+- Visual Studio 2019 with the **.NET Core cross-platform development** workload
+- [.NET SDK 5.0](https://dotnet.microsoft.com/download/dotnet/5.0) or later
```
Comment (scottaddie) [inline WhatsNew.Cli/CONTRIBUTING.md:6]: Should these prereqs be updated to newer versions?

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    include:
+      - whatsnew/*
+    exclude:
+      - whatsnew/src/WhatsNew.Cli/CONTRIBUTING.md
+      - whatsnew/src/WhatsNew.Cli/README.md
+```
+
+With the preceding configuration:
+
+- A commit to *master* or a branch prefixed with *scottaddie/* triggers the build.
```
Comment (scottaddie) [inline WhatsNew.Cli/CONTRIBUTING.md:87]:
```suggestion
- A commit to *main* or a branch prefixed with *scottaddie/* triggers the build.
```

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+- Any file in the *whatsnew* directory triggers a build, with the exception of the Markdown files listed in `exclude`.
+
+### Release
+
+Publishing of the NuGet package (containing the .NET global tool) is handled by the last task in the build pipeline. The package version number is determined by the *WhatsNew.Cli* project's `<Version>` element's value. If the publish is successful, the new package version appears in the Azure Artifacts [DotnetDocsTools feed](https://dev.azure.com/mseng/TechnicalContent/_packaging?_a=feed&feed=DotnetDocsTools%40Local).
+
+Complete the following steps to publish the package:
+
+1. Increment the version number in the project file's `<Version>` element and commit the change. Wait for the build to complete, and ensure that the change has been merged to *master*.
+1. Select the **Run pipeline** button at https://dev.azure.com/mseng/TechnicalContent/_build?definitionId=10534.
```
Comment (scottaddie) [inline WhatsNew.Cli/CONTRIBUTING.md:97]: Is this dev.azure.com link still valid with the move to the public repo?

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+- A commit to *master* or a branch prefixed with *scottaddie/* triggers the build.
+- Any file in the *whatsnew* directory triggers a build, with the exception of the Markdown files listed in `exclude`.
+
+### Release
+
+Publishing of the NuGet package (containing the .NET global tool) is handled by the last task in the build pipeline. The package version number is determined by the *WhatsNew.Cli* project's `<Version>` element's value. If the publish is successful, the new package version appears in the Azure Artifacts [DotnetDocsTools feed](https://dev.azure.com/mseng/TechnicalContent/_packaging?_a=feed&feed=DotnetDocsTools%40Local).
+
+Complete the following steps to publish the package:
+
+1. Increment the version number in the project file's `<Version>` element and commit the change. Wait for the build to complete, and ensure that the change has been merged to *master*.
```
Comment (scottaddie) [inline WhatsNew.Cli/CONTRIBUTING.md:96]:
```suggestion
1. Increment the version number in the project file's `<Version>` element and commit the change. Wait for the build to complete, and ensure that the change has been merged to *main*.
```

File: WhatsNew.Cli/README.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+Complete the following steps in Visual Studio Code:
+
+1. Create a JSON configuration file within the GitHub repository:
+    - For docsets in the *azure-docs-pr* repository, the file's naming convention is *.\{SERVICE-NAME}.json*. The file should live in the repository's *.whatsnew* directory. Replace the *\{SERVICE-NAME}* placeholder with the directory name within *articles* in which the docs are stored. For example, name the file *.app-service.json* for Azure App Service docs.
+    - For all other docsets, the file should be named *.whatsnew.json* and live in the repository's root directory.
+1. Associate your JSON configuration file with the JSON schema by adding a `$schema` property to line 2 of the JSON configuration file. The `$schema` property's value should be set as follows:
+
+    ```json
+    "$schema": "https://whatsnewapi.azurewebsites.net/schema",
```
Comment (scottaddie) [inline WhatsNew.Cli/README.md:13]: BTW, this API is no longer available. It was an ASP.NET Core web API running in App Service that provided the JSON schema. You'd need this `$schema` line to get IntelliSense in your JSON file. Let me know if you have an Azure subscription we can use, and I can get this going again.

File: WhatsNew.Cli/CONTRIBUTING.md (hunk 5 above)
Comment (BillWagner) [reply WhatsNew.Cli/CONTRIBUTING.md:97]: No, but I'll update that in a later PR. (We plan to update this to make it a GitHub Action. The action will pull the source and build it.)

File: WhatsNew.Cli/README.md (hunk 7 above)
Comment (BillWagner) [reply WhatsNew.Cli/README.md:13]: Would it be better to put the schema here in this repo? It's a public URL.

File: WhatsNew.Cli/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+### Release
+
+Publishing of the NuGet package (containing the .NET global tool) is handled by the last task in the build pipeline. The package version number is determined by the *WhatsNew.Cli* project's `<Version>` element's value. If the publish is successful, the new package version appears in the Azure Artifacts [DotnetDocsTools feed](https://dev.azure.com/mseng/TechnicalContent/_packaging?_a=feed&feed=DotnetDocsTools%40Local).
+
+Complete the following steps to publish the package:
+
+1. Increment the version number in the project file's `<Version>` element and commit the change. Wait for the build to complete, and ensure that the change has been merged to *main*.
+1. Select the **Run pipeline** button at https://dev.azure.com/mseng/TechnicalContent/_build?definitionId=10534.
+1. Select *master* as the branch in the **Branch/tag** drop down list.
```
Comment (IEvangelist) [inline WhatsNew.Cli/CONTRIBUTING.md:98]:
```suggestion
1. Select *main* as the branch in the **Branch/tag** drop down list.
```

File: WhatsNew.Infrastructure.Tests/Services/ConfigurationServiceTests.cs [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    [InlineData("  ")]
+    [InlineData(null)]
+    public async Task Empty_GitHub_Personal_Access_Token_Throws_InvalidOperationException(
+        string gitHubKey)
+    {
+        var input = new PageGeneratorInput
+        {
+            Owner = "dotnet",
+            Repository = "AspNetCore.Docs",
+            Branch = "master",
```
Comment (IEvangelist) [inline WhatsNew.Infrastructure.Tests/Services/ConfigurationServiceTests.cs:47]:
```suggestion
            Branch = "main",
```

File: WhatsNew.Infrastructure.Tests/Services/ConfigurationServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    [Fact(Skip = "Flaky in Az Pipelines build")]
+    public async Task Valid_Org_And_Repo_With_Invalid_DocSet_Throws_FileNotFoundException()
+    {
+        var input = new PageGeneratorInput
+        {
+            Owner = "dotnet",
+            Repository = "docs",
+            Branch = "master",
```
Comment (IEvangelist) [inline WhatsNew.Infrastructure.Tests/Services/ConfigurationServiceTests.cs:23]:
```suggestion
            Branch = "main",
```

File: WhatsNew.Cli/README.md (hunk 7 above)
Comment (scottaddie) [reply WhatsNew.Cli/README.md:13]: That could work.

**Pull request merged**